src/materials/PhysicalPathTracingMaterial.js

```diff
@@ -660,7 +660,7 @@ export class PhysicalPathTracingMaterial extends MaterialBase {
 								// get the material pdf
 								vec3 sampleColor;
 								float lightMaterialPdf = bsdfResult( outgoing, normalize( invBasis * lightSampleRec.direction ), surfaceRec, sampleColor );
-								if ( lightMaterialPdf > 0.0 ) {
+								if ( lightMaterialPdf > 0.0 && ! any( isnan( sampleColor ) ) ) {
```

Comment:
I'll see if I can get rid of the warnings by moving things around a bit. Observing the difference with and without the check make it abundantly clear that there are in fact NaN values at times. The warning itself is also quite funnily worded IMO. It _cannot_ be NaN, yet `isnan()` _may_ not be necessary.

Reply:
Okay, I have the feeling ANGLE is doing some aggressive inlining. Placing the `isnan` check all the way into `specularColor` prevents the warnings, but makes communicating the result back quite ugly. Instead I've decided to not check for NaN values, but instead check the validity of the `sampleColor` by checking that all components (rgb) are >= 0.0, since we know negative color shouldn't occur.
```glsl
bool isValidSampleColor = all( greaterThanEqual( sampleColor, vec3( 0.0 ) ) );
```

Reply:
It is a fairly odd warning... but thanks for taking a look at it! I was just as happy chalking it up to an ANGLE quirk!